### PR TITLE
Bump kube-state-metrics to v1.7.0

### PR DESCRIPTION
**Pull request by metalmatze**
Let's wait until v1.7.0 with merging.

### CODE REVIEW

**Comment by brancz** [comment]
happy to merge this and get people testing it

**Comment by metalmatze** [comment]
Ok! :+1:

File: jsonnet/kube-prometheus/kube-state-metrics/kube-state-metrics.libsonnet [hunk 1]
```diff
@@ -16,7 +16,7 @@ local k = import 'ksonnet/ksonnet.beta.4/k.libsonnet';
     },
 
     versions+:: {
-      kubeStateMetrics: 'v1.6.0',
+      kubeStateMetrics: 'v1.7.0-rc.1',
```
Comment (LiliC) [inline jsonnet/kube-prometheus/kube-state-metrics/kube-state-metrics.libsonnet]:
```suggestion
      kubeStateMetrics: 'v1.7.1',
```

We should use that, as it includes some bug fixes https://github.com/kubernetes/kube-state-metrics/releases/tag/v1.7.1

**Comment by LiliC** [comment]
I had to push some commits @metalmatze to this PR, as we were seeing errors in kube-state-metrics logs.

cc @s-urbaniak as discussed offline PTAL :) Thanks!

File: jsonnet/kube-prometheus/kube-state-metrics/kube-state-metrics.libsonnet [hunk 2]
```diff
@@ -43,88 +43,95 @@ local k = import 'ksonnet/ksonnet.beta.4/k.libsonnet';
...
+        ]) +
+        rulesType.withVerbs(['list', 'watch']),
+
+        rulesType.new() +
+        rulesType.withApiGroups(['storage.k8s.io']) +
+        rulesType.withResources([
+          'storageclasses',
+        ]) +
+        rulesType.withVerbs(['list', 'watch']),
+      ];
```
Comment (s-urbaniak) [inline jsonnet/kube-prometheus/kube-state-metrics/kube-state-metrics.libsonnet]: this looks super clean :+1:

**Pull request merged**